amethyst_assets/src/storage.rs

```diff
@@ -230,10 +332,46 @@ where
     type Storage = DenseVecStorage<Self>;
 }
 
-// TODO: may change with hot reloading
-pub struct Processed<A: Asset> {
-    pub data: Result<A::Data, BoxedErr>,
-    pub format: String,
-    pub handle: Handle<A>,
-    pub name: String,
+pub enum Processed<A: Asset> {
```

Comment:
It `pub` necessary here?

Reply:
Enums can't have private members.

Reply:
Sorry, I think I misunderstood your question. Yes, `Processed` needs to be pub, because `Loader` creates such a structure. It's not exported at crate root, though.

Reply:
Can you make it `pub(crate)` then?

Reply:
There's no need for that and besides, I think that syntax is horrible. I try not to use it where possible.

Reply:
Why?  It's very straightforward and obvious.  You don't need to look at other files to figure out it's not actually pub, and pub(crate) permits easy re-use of internal structures.